test/functional/api/vim_spec.lua

```diff
@@ -853,6 +853,61 @@ describe('API', function()
         feed('u') -- Undo.
         expect(expected1)
       end)
+      it('stream: multiple chunks sets the correct mark [', function()
+        api.nvim_paste('1/chunk 1 (start)\n', true, 1)
+        eq({ 0, 1, 1, 0 }, fn.getpos("'["))
+        api.nvim_paste('1/chunk 2\n', true, 2)
+        eq({ 0, 2, 1, 0 }, fn.getpos("'["))
+        api.nvim_paste('1/chunk 3 (end)\n', true, 3)
+        local expected1 = [[
+          1/chunk 1 (start)
+          1/chunk 2
+          1/chunk 3 (end)
+          ]]
+        expect(expected1)
+        eq({ 0, 1, 1, 0 }, fn.getpos("'["))
+        api.nvim_paste('aaaaaa', true, -1)
+        eq({ 0, 4, 1, 0 }, fn.getpos("'["))
+        api.nvim_paste('bbbbbb', true, 1)
+        eq({ 0, 4, 7, 0 }, fn.getpos("'["))
+        api.nvim_paste('cccccc', true, 2)
+        eq({ 0, 4, 13, 0 }, fn.getpos("'["))
+        api.nvim_paste('dddddd\n', true, 3)
+        eq({ 0, 4, 7, 0 }, fn.getpos("'["))
+        expected1 = [[
+          1/chunk 1 (start)
+          1/chunk 2
+          1/chunk 3 (end)
+          aaaaaabbbbbbccccccdddddd
+          ]]
+        expect(expected1)
+        -- Only pastes an empty line
+        api.nvim_paste('', true, -1)
+        eq({ 0, 5, 1, 0 }, fn.getpos("'["))
+        api.nvim_paste('', true, 3)
```

Comment:
This chunk has paste `3`, but it doesn't have a corresponding chunk with phase `1`.